First pass at addng GENIE info

krwood opened the pull request:
Added the GENIE header/summary information to the ndflow output

References between GENIE interactions and tracks/segments and the trajectories seems to have worked.

Applied same unit and axes conversions to the GENIE interaction vertex to match the trajectories.

Modified how the truth info slices were calculated to be uniform, fix at least one bug, and to be simpler (in my opinion). The original methods are left in as commented code.

cuddandr committed:
First pass at addng GENIE info

Added the GENIE header/summary information to the ndflow output

References between GENIE interactions and tracks/segments and the
trajectories seems to have worked.

Applied same unit and axes conversions to the GENIE interaction vertex
to match the trajectories.

Modified how the truth info slices were calculated to be uniform, fix at least one
bug, and to be simpler (in my opinion). The original methods are left in
as commented code.

krwood commented on src/proto_nd_flow/reco/charge/raw_event_generator.py:
```diff
@@ -154,17 +154,25 @@ def _convert_mc_truth_tracks(self, tracks):
-            traj_copy[field][:, 0] = traj[field][:, 2] * units.MeV
-            traj_copy[field][:, 1] = traj[field][:, 1] * units.MeV
-            traj_copy[field][:, 2] = traj[field][:, 0] * units.MeV
+            traj_copy[field][:, 0] = traj[field][:, 2] * units.keV
```
What is the rationale for keV instead of MeV here?

cuddandr replied:
Running ndflow with the units set to MeV for the trajectories gives output that is 1000x too large for the momentum (e.g. pxyz_start). The attached plot shows the muon initial momentum distribution and note the axis scale, the peak is at ~5E6 MeV or 5 TeV. This is present in the latest batch of MiniRun3 Flow files and from running ndflow myself over the MiniRun3 larndsim files.

Changing the units to keV fixes this, however looking at the larndim files the trajectory momentum seems to be correctly in MeV... so I'm not sure if there is some other conversion being applied to the units.

![image](https://user-images.githubusercontent.com/33762574/233474663-a560a736-2acb-4f1d-8742-ae82f12af48e.png)

cuddandr replied:
I will also note here (so as not to forget) that I need to double check the y-offset and the position units for trajectories, and if the GENIE vertex needs the y-offset applied.

cuddandr committed:
Setting MC truth units to MeV and cm

Removing the unit conversions for energy and distance in the convert
dataset functions. This keeps the same units as the input from larndsim,
which are MeV and cm for energy and distance respectively.

Original convention was to convert to the default units for ndflow,
which are keV and mm.

krwood commented:
@mjkramer @cuddandr Should we merge this into MiniRun3_dev?

krwood commented:
Nevermind, I see this was already done https://github.com/DUNE/ndlar_flow/commit/cb8594401eaa43cbff68cb69634a20f3a66c56ec

I'm just going to delete this branch since we have it in MiniRun3 (which we should merge into develop at some point)
